Analysis of seer flag (#7)
BZFlag-Dev/bzflag.org: _documentation/user/flags/seer.md

summary: See stealthed, cloaked and masquerading tanks as normal.
---

> The Seer (SE) flag allows the player to see the tanks holding the [Stealth (ST)](../stealth/), [Cloaking (CL)](../cloaking/), or [Masquerade (MQ)](../masquerade/) flags as normal.

Reviewer: ```suggestion
The Seer (SE) flag allows the player to see through tanks camouflaging themselves or their shots either from in the window or radar of other players. This flag nullifies the effects of the following flags for the beholder:

- [Cloaking (CL)](../cloaking/)
- [Invisible Bullet (IB)](../invisible-bullet/)
- [Masquerade (MQ)](../masquerade/)
- [Stealth (ST)](../stealth/)
```

Possibly something like this instead?

> Its power is therefore to nullify the effects of those three flags for the holder, rather than bringing out a unique power-up itself.

Reviewer: ```suggestion
Its power is therefore to nullify the effects of those flags for the holder, rather than bringing out a unique power-up itself.
```

While unlikely this would happen any time soon, I'd prefer to avoid specifying set limits as new flags could be introduced in future versions of the game

Author: @allejo yeah not mentioning the count would make it more flexible, agreed.

## Tactics

> As the flag disables the effect of three other specific flags for the holder, the usage of Seer is highly situational.

Reviewer: I'd avoid specifying a hard limit since updating this value (if ever) could be hard to remember

> In general, a player would prefer a flag that gives a power-up itself rather than nullify that of some others, as a power-up can effect most players on the map, while Seer does not add any effect when battling against tanks that have flags other than the three affected by Seer.

Reviewer: ```suggestion
A player may prefer a flag that gives a power-up itself rather than nullify that of some others, as a power-up can effect most players on the map, while Seer does not add any effect when battling against tanks that have camouflaging flags.
```

No hard count and slightly more objective?

Reviewer 2: why is a preference part of tactics? Focus on how to use to advantage, particularly finding a pesky stealth tank (think Rabbit Hunt)

Reviewer 2: I'm just saying, "A player would (may) prefer" is a preference, not a tactic. All (non-bad) flags have some value. It seems to me the point is to describe the value.

Author: Then it can be worded as 'A player prefers' if it's such a sensitive wording, or perhaps swapping the word 'prefers' out for some other completely.

The point is to describe the tactics, not value, which seems, at least to me, when to and when not to use the flag (in relation to other flags).

I believe this line that you deem as preference is actually the most crucial one, as newer players can easily understand the effect, but not the tactics. They ask themselves 'I just grabbed this flag, should I keep it?'. And this line examines the situation of when it is a good tactic to keep it and, of course, when it is not.

And it isn't preference, it is effectiveness. Or, you could call it preference towards effectiveness, which is usually what one wants.

Author: @allejo @jwmelo in accordance to the post above, we ought to decide whether we will comprehend the 'tactics' field as flag 'value' (meaning positives only) or flag 'situational usage in relation to other flags' (meaning positives and negatives, or better worded, when to and when not to use a flag).

Author: @allejo @jwmelto in accordance to the post above, we ought to decide whether we will comprehend the 'tactics' field as flag 'value' (meaning positives only) or flag 'situational usage in relation to other flags' (meaning positives and negatives, or better worded, when to and when not to use a flag).

Reviewer: Being the section is titled "Tactics" and has been standardized amongst all the flags, the section should be dedicated to actual tactics of the flag; i.e. how to use the flag and when or what strategies exist.

As for preferences/opinion, I agree that we shouldn't have that. We can have statements like this in the docs, but just the wording of it shouldn't imply opinions or preferences.

Author: I think we all agree on that.

Reviewer 2: I don't want to go back and forth on this. Don't compare this flag to other flags, and why you would want another one over this one. Describe how to use this flag effectively. Period.

Author: Where did I ever compare this flag to another? If you are meaning to pose arguments, back them up. I wrote about this flag's strength is certain situations and weaknesses in others. I never mentioned any other specific flags, just all other flags as a collective group, which I believe is a matter very worth bringing up in the tactics field of this specific flag to examine its usage entirely.

Reviewer 2: > In general, a player would prefer a flag that gives a power-up itself rather than nullify that of some others, as a power-up can effect most players on the map, while Seer does not add any effect when battling against tanks that have flags other than the three affected by Seer.

This does not describe tactics, but is explicitly a point of preference. It is, and has always been, the topic of this extensive thread. Just remove the sentence.

Author: Or even *less* mentioning of other flags:

> Seer does not add any effect when battling against tanks that have flags other than those affected by Seer.

> As such, the usage is usually limited to the situations where either a holder of one of the three affected flags is in close proximity to the player holding Seer, or the majority of the tanks on the map are holding one of the three affected flags.

Reviewer: I'd opt towards talking about how Seer can be useful when hunting down rabbits with stealth or pesky users always shooting you in the back. I like the mention of this flag being powerful on maps with a lot of stealth.

Reviewer 2: yeah ... great idea!

Author: @allejo @jwmelto thank you for the comments. While there can be a mention of finding stealth rabbits, I thought it would be redundant, as that isn't any more major than a teammate locating a stealth tank incoming towards the base on CTF and letting everyone on the team know, or a teammate (or even often times an enemy) locating a stealth hiding in a dangerous location on FFA and letting everyone on the map know.

We can mention these cases, but as they more or less span across all game modes, I thought it wouldn't make sense to have preferences towards one.

Reviewer: The Seer flag doesn't introduce special effects like other flags, it instead nullifies the effect of another flag. For this reason, I think it'd be beneficial to discuss which tactics this flag can defeat since that's essentially the power of this flag. Rabbits using stealth is a fairly common one, a stealth-mostly map is another, a stealth camping in a base, etc.

Author: As said, we can list all. It might end up quite extensive, but it is better than discriminating some situations over others.
